components/auth/org.wso2.carbon.auth.scim.rest.api/src/main/java/org/wso2/carbon/auth/scim/rest/api/impl/UsersApiServiceImpl.java

```diff
@@ -1,57 +1,107 @@
 
 public class UsersApiServiceImpl extends UsersApiService {
+    private static final Log LOG = LogFactory.getLog(UsersApiServiceImpl.class);
+
     @Override
-    public Response usersGet(Integer startIndex
-, Integer count
-, String filter
-  ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response usersGet(Integer startIndex, Integer count, String filter, Request request)
+                                                                                    throws NotFoundException {
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            UserResourceManager userResourceManager = new UserResourceManager();
+            SCIMResponse scimResponse = userResourceManager.listWithGET(userManager, filter, startIndex, count,
+                    null, null, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
+
     @Override
-    public Response usersIdDelete(String id
-  ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response usersIdDelete(String id, Request request) throws NotFoundException {
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            UserResourceManager userResourceManager = new UserResourceManager();
+            SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
+
     @Override
-    public Response usersIdGet(String id
-  ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response usersIdGet(String id, Request request) throws NotFoundException {
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            UserResourceManager userResourceManager = new UserResourceManager();
+            SCIMResponse scimResponse = userResourceManager.get(id, userManager, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
```

Comment:
Exception has not been logged. Need to log exception as follows,

log.error("Message", e);